src/sfizz/Region.cpp

```diff
@@ -1751,22 +1769,37 @@ float sfz::Region::getDelay() const noexcept
     return Default::delay.bounds.clamp(finalDelay);
 }
 
-uint32_t sfz::Region::trueSampleEnd(Oversampling factor) const noexcept
+uint32_t sfz::Region::getSampleEnd(Oversampling factor) const noexcept
 {
-    if (sampleEnd <= 0)
-        return 0;
+    int64_t end = sampleEnd;
+    for (const auto& mod: endCC)
+        end += static_cast<int64_t>(mod.data * midiState.getCCValue(mod.cc));
+
+    end = max(int64_t { 0 }, end);
```

Comment:
not clamped in both directions?

Reply:
You mean at sample end?

Reply:
Yes, if needed and not done already somewhere else